src/coreclr/jit/llvm.h

```diff
@@ -350,19 +342,38 @@ class Llvm
     GenTree* normalizeStructUse(LIR::Use& use, ClassLayout* layout);
 
     unsigned representAsLclVar(LIR::Use& use);
-    GenTree* createStoreNode(var_types nodeType, GenTree* addr, GenTree* data);
-    GenTree* createShadowStackStoreNode(var_types storeType, GenTree* addr, GenTree* data);
     GenTree* insertShadowStackAddr(GenTree* insertBefore, ssize_t offset, unsigned shadowStackLclNum);
 
+    unsigned getCatchArgOffset() const;
+
+    // ================================================================================================================
+    // |                                           Shadow stack allocation                                            |
+    // ================================================================================================================
+public:
+    void Allocate();
+
+private:
+    void allocSpillTempsLiveAcrossSafePoints();
+
+    void allocInitializeAndAllocateLocals();
+    void allocDissolvePromotedLocals();
+    void allocAssignShadowFrameOffsets(std::vector<unsigned>& shadowFrameLocals);
+
+    void allocLowerAndInsertProlog();
+    void allocInitializeLocalInProlog(unsigned lclNum, GenTree* value);
+
+    void allocRewriteShadowFrameReferences();
+    void allocRewriteLocal(GenTreeLclVarCommon* lclNode);
+    void allocRewriteCall(GenTreeCall* call);
+
```

Comment:
I expect that this code structure is more or less temporary and LSSA proper will be put into its own class.